packages/block-library/src/image/edit.js

```diff
@@ -108,11 +108,35 @@ export function ImageEdit( {
 		align,
 		metadata,
 	} = attributes;
+
 	const [ temporaryURL, setTemporaryURL ] = useState( attributes.blob );
 
 	const [ contentResizeListener, { width: containerWidth } ] =
 		useResizeObserver();
 
+	const [ placeholderResizeListener, { width: placeholderWidth } ] =
+		useResizeObserver();
+
+	// Parse width of the image block attribute.
+	const parseWidth = ( widthValue ) => {
+		const parsed = parseInt( widthValue, 10 );
+		return isNaN( parsed ) ? null : parsed;
+	};
+
+	const parsedWidth = parseWidth( width );
+
+	// Get the effective width of the image placeholder use the minimum of placeholder width
+	// or image attribute width if both are defined. If only one is defined, use that one.
+	const getEffectiveWidth = () => {
+		if ( placeholderWidth !== null && parsedWidth !== null ) {
+			return Math.min( placeholderWidth, parsedWidth );
+		}
+		return placeholderWidth !== null ? placeholderWidth : parsedWidth;
+	};
+
+	const effectiveWidth = getEffectiveWidth();
+	const isLargeContainer = effectiveWidth !== null && effectiveWidth > 160;
```

Comment:
```suggestion
	const isLargeContainer = placeholderWidth > 160;
```

Do we need to take into account the width as an attribute? I might be missing something, but can't we simply change it like this?

Reply:
Thanks @t-hamano I was actually using the observer outside the placeholder component that's why it was required earlier but if I move inside it then I don't think we will need that. So I will update the PR accordingly.